### ISSUE TITLE
[CAL-2109] Reschedule Emails not received by the Scheduler

### ISSUE THREAD

**Issue by alishaz-polymath** [calcom/cal.com#9957]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

I had scheduled a couple of meetings yesterday. Both the meetings were rescheduled by the Hosts. The meetings updated in the /bookings section of [cal.com](http://cal.com) app, however, I didn't receive an email for either of them.
Another point of note is, the previous calendar entry from my google calendar was removed (which is good and expected) but the rescheduled event didn't show up on the calendar.

In terms of opt-in bookings, when the scheduler reschedules the meeting before it has been accepted, they now have two opt-in meetings instead of removing the old one, which is the expected outcome.

Destination Calendar: Google

### Steps to Reproduce

1. Schedule a booking
2. Ask host to reschedule it

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Technical details

* Production

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2109](https://linear.app/calcom/issue/CAL-2109/reschedule-emails-not-received-by-the-scheduler)</sub>

**Comment by shirazdole** [comment]
Another issue with rescheduling is for opt-in meetings, when a user "requests" an opt-in meeting, they're still able to reschedule that meeting within the email sent to them. 

However, when they select a new time, the previous requested booking is still valid and not canceled. That makes it confusing for the hosts b/c as the host I think I have two options to approve from, but in reality, I only have 1 because the first request was "rescheduled" or "canceled" — the ideal behavior here is to cancel the first requested booking so that when the host goes to accept it — it says "this request has been canceled," but I'm not 100% on this.

**Comment by alishaz-polymath** [comment]
> Another issue with rescheduling is for opt-in meetings, when a user "requests" an opt-in meeting, they're still able to reschedule that meeting within the email sent to them.
> 
> However, when they select a new time, the previous requested booking is still valid and not canceled. That makes it confusing for the hosts b/c as the host I think I have two options to approve from, but in reality, I only have 1 because the first request was "rescheduled" or "canceled" — the ideal behavior here is to cancel the first requested booking so that when the host goes to accept it — it says "this request has been canceled," but I'm not 100% on this.

Agreed! It should cancel that. Updating the above with this info 🙏

**Comment by iamr-kumar** [comment]
If no one has taken up this yet, can I work on this?

**Comment by keithwillcode** [comment]
Just assigned you @iamr-kumar. Appreciate your willingness to jump in. This one is Urgent so excited to see your approach. 🙏

**Comment by iamr-kumar** [comment]
Thanks @keithwillcode. I am having some trouble in setting up the calendar in the dev environment. For example, not able to connect to Google calendar from the onboarding steps. Is there something I could be missing?

**Comment by iamr-kumar** [comment]
Hey @alishaz-polymath, I am not able to reproduce this bug both locally and on prod. I tried rescheduling the meeting from the host's side and I am getting the rescheduled email and also the entries on Google calendar is updating properly (where the old one is deleted and new event is created).

Could you help me with any more details to reproduce this issue?

**Comment by alishaz-polymath** [comment]
@shirazdole Can you provide the steps for the issue @iamr-kumar mentions here?

**Comment by iamr-kumar** [comment]
Hey @shirazdole, is there any update on this as I still have not been able to reproduce it? Does this issue still persist as @keithwillcode had marked it as urgent. Thanks

**Comment by nicktrn** [comment]
I have also just tried to reproduce these issues and couldn't. Seems to work just fine with linked Google Calendar and without.

Tested scenarios:
- reschedule normal meeting
- reschedule opt-in meeting
- same again but without linked calendar

Outcome:
- all emails arrive as they should
- all calendar entries are created and updated as expected
- there are no duplicate unconfirmed /bookings

The issue should probably be closed, unless @alishaz-polymath or @shirazdole can confirm this is still happening.
My guess is this was either fixed or only happens under very specific circumstances.

Could have something to do with #10156. Currently, failure to delete calendar events during rescheduling does not prevent reschedule emails being sent out. See: [requestReschedule.handler.ts:198](https://github.com/calcom/cal.com/blob/d55404b3aa032588a660b0c594775ff75a396d8e/packages/trpc/server/routers/viewer/bookings/requestReschedule.handler.ts#L198)

As I've mentioned in the above issue and linked PR #10157 it's very hard to predict what this can result in, but rescheduling is definitely affected.
